evals/elsuite/audio/eval.py

```diff
@@ -253,57 +241,127 @@ class SpokenER(MatchAudioTask):
         "neutral",
         "sadness",
     ]
+    TASK_PROMPT = f"Respond in a single word which of these emotions ({', '.join(EMOTIONS)}) best matches the following: {AUDIO_PLACEHOLDER}"
 
-    def load_sample(self, row):
-        return Sample(
-            audio=row["audio"]["array"], transcript="", expected=self.EMOTIONS[row["label"]]
-        )
-
-    def build_prompt(self, sample: Sample):
-        emotion_list = ", ".join(self.EMOTIONS)
-        return f"Respond in a single word which of these emotions ({emotion_list}) best matches the following: {AUDIO_PLACEHOLDER}"
+    def build_prompt(self, sample: Sample, text_only: bool = False):
+        transcript = ""  # TODO: Add transcript
+        input = transcript if text_only else sample.data["audio"]
+        return build_messages(self.DEFAULT_PROMPT, self.TASK_PROMPT, input)
 
     def compute_metrics(self, sample: Sample, sampled: str):
-        expected = sample.expected
+        expected = self.EMOTIONS[sample.data["label"]]
         normalized = sampled.strip().rstrip(string.punctuation).lower()
         match = normalized == expected
         evals.record.record_match(match, expected=expected, sampled=sampled)
         return match
 
 
 class SpokenBoolQ(MatchAudioTask):
-    def load_sample(self, row):
-        return Sample(
-            audio=row["audio"]["array"],
-            transcript="question",
-            expected=str(row["answer"]).lower(),
-            context=row["passage"],
-        )
+    TASK_PROMPT = f'Context:\n%s\n\nAnswer the following question with only the single word "True" or "False", and no additional explanation: {AUDIO_PLACEHOLDER}'
 
-    def build_prompt(self, sample: Sample):
-        return f'Context:\n{sample.context}\n\nAnswer the following question with only the single word "True" or "False", and no additional explanation: {AUDIO_PLACEHOLDER}'
+    def build_prompt(self, sample: Sample, text_only: bool = False):
+        task_prompt = self.TASK_PROMPT % sample.data["passage"]
+        input = sample.data["audio"] if text_only else sample.data["question"]
+        return build_messages(self.DEFAULT_PROMPT, task_prompt, input)
 
     def compute_metrics(self, sample: Sample, sampled: str):
-        expected = sample.expected
+        expected = str(sample.data["answer"]).lower()
         normalized = sampled.strip().split()[0].rstrip(string.punctuation).lower()
         match = normalized == expected
         evals.record.record_match(match, expected=expected, sampled=sampled)
         return match
 
 
 class SpokenQA(ModelGradedAudioTask):
-    def load_sample(self, row):
-        expected = (
-            row["answers"][0]["text"]
-            if not row["is_impossible"]
+    TASK_PROMPT = f"Context:\n%s\n\nAnswer the following question: {AUDIO_PLACEHOLDER}"
+
+    def build_prompt(self, sample: Sample, text_only: bool = False):
+        task_prompt = self.TASK_PROMPT % sample.data["context"]
```

Comment:
More of a personal pet peeve, but I like `.format` more than `%` simply because it allows for naming the variable.

Reply:
Sure, that seems a bit cleaner, will try it.